src/unix/thread.c

```diff
@@ -51,6 +51,60 @@ sysreturn arch_prctl(int code, unsigned long addr)
 }
 #endif
 
+struct clone_args_internal {
+     u64 flags;
+     int *child_tid;
+     int *parent_tid;
+     void *stack;
+     bytes stack_size;
+     u64 tls;
+};
+
+sysreturn clone_internal(struct clone_args_internal *args, bytes size)
+{
+     u64 flags = args->flags;
+
+     if (!args->stack_size)
+          return set_syscall_error(current, EINVAL);
+
+     if (!(flags & CLONE_THREAD)) {
+          thread_log(current, "attempted to create new process, aborting.");
+          return set_syscall_error(current, ENOSYS);
+     }
+
+     if (!validate_user_memory(args->stack, args->stack_size, true))
```

Comment:
Since args->stack points to the top of the stack, this validation should be done from (args->stack - args->stack_size) to args->stack

Reply:
I switched this around again so that clone_internal expects the bottom of the stack. Thanks!